airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/MySqlSource.java

```diff
@@ -229,4 +238,42 @@ public static void main(final String[] args) throws Exception {
     };
   }
 
+  private void checkBinlog(JsonNode offset, JdbcDatabase database) {
+    String binlog = getBinlog(offset);
+    if (binlog != null && !binlog.isEmpty()) {
+      if (isBinlogAvailable(binlog, database)) {
+        LOGGER.info(String.format("Binlog %s is available", binlog));
+      } else {
+        String error =
+            String.format("Binlog %s is not available. This is a critical error, it means that requested binlog is not present on mysql server. " +
```

Comment:
I believe, we can use Text block here instead of string concatenation https://openjdk.java.net/jeps/378, as we migrate to Java 17.